pkg/middlewares/headers/headers.go

```diff
@@ -110,64 +109,74 @@ func (s secureHeader) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 // Header is a middleware that helps setup a few basic security features. A single headerOptions struct can be
 // provided to configure which features should be enabled, and the ability to override a few of the default values.
 type Header struct {
-	next    http.Handler
-	headers *config.Headers
+	next             http.Handler
+	hasCustomHeaders bool
+	hasCorsHeaders   bool
+	headers          *config.Headers
 }
 
 // NewHeader constructs a new header instance from supplied frontend header struct.
 func NewHeader(next http.Handler, headers config.Headers) *Header {
+	hasCustomHeaders := headers.HasCustomHeadersDefined()
+	hasCorsHeaders := headers.HasCorsHeadersDefined()
+
 	return &Header{
-		next:    next,
-		headers: &headers,
+		next:             next,
+		headers:          &headers,
+		hasCustomHeaders: hasCustomHeaders,
```

Comment:
these two are just conveniences, right? we could simply call Header.headers.HasCustomHeadersDefined() everywhere instead, couldn't we?

Reply:
Yes, but I do not want to check if custom headers are present in the configuration on every request. Check once, and save the result. Speeds up processing :)

Reply:
ACK